### PR TITLE
NodeJS Week3 Homework

### PR DESCRIPTION
**Pull request by slastushevskyi**

### CODE REVIEW

File: src/backend/api/meals.js [hunk 1]
```diff
@@ -2,13 +2,128 @@ const express = require("express");
...
+          res.json(await mealsWithRemainingReservations);
+        }
+        // Returns all meals that partially match the given title.
+        // Rød grød will match the meal with the title Rød grød med fløde.
+      } else if (key === "title") {
+        const meals = await knex
+          .select()
+          .from("Meal")
+          .where("title", "like", `%${value}%`);
+        res.status(200).json(await meals);
```
Comment (itsjrm) [inline src/backend/api/meals.js]: No await here.

File: src/backend/api/meals.js [hunk 2]
```diff
@@ -2,13 +2,128 @@ const express = require("express");
...
+      } else if (key === "dateBefore") {
+        const meals = await knex
+          .select()
+          .from("Meal")
+          .where("when", "<", `${value}`);
+        res.status(200).json(await meals);
+        // Returns the given number of meals.
+      } else if (key === "limit") {
+        const meals = await knex.select().from("Meal").limit(`${value}`);
+        res.status(200).json(await meals);
```
Comment (itsjrm) [inline src/backend/api/meals.js:151]: No await here.

File: src/backend/api/reviews.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  }
+});
+
+//POST Adds a new review to the database
+router.post("/", async (req, res) => {
+  try {
+    const newReview = req.body;
+    const [id] = await knex("Review").insert(newReview);
+    if ([id]) {
+      res.status(201).json({
```
Comment (itsjrm) [inline src/backend/api/reviews.js]: No need to put `id` back into an array after destructuring.

**Review by itsjrm — APPROVED** [review]
Did you see my comments on double awaits?

File: src/backend/api/meals.js (hunk 1 above)
Comment (slastushevskyi) [reply src/backend/api/meals.js]: Yes, yes, I forgot to remove it :)

**Pull request merged**